rules/no-reduce.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+const methodSelector = require('./utils/method-selector');
+const getDocumentationUrl = require('./utils/get-documentation-url');
+
+const message = 'Array.reduce not allowed';
+
+const PROTOTYPE_SELECTOR = [
+	methodSelector({names: ['call', 'apply']}),
```

Comment:
Please split call and apply,

 and limit call arguments to 2~3 and apply arguments to 2

Reply:
Why should this be split? `methodSelector` already takes care if `names` is an `array`.

Reply:
the selector is already narrowed and will only hit for `[].call.reduce` and `Array.prototype.call.reduce`, which means the code is using "reduce", shouldn't the rule be triggered irrespective of the arg length?

Reply:
Okay, It's strict enough, let's keep it.